sqlmesh/utils/concurrency.py

```diff
@@ -13,9 +13,32 @@
 
 
 class NodeExecutionFailedError(t.Generic[H], SQLMeshError):
-    def __init__(self, node: H):
+    def __init__(self, ex: Exception, node: H):
         self.node = node
-        super().__init__(f"Execution failed for node {node}")
+
+        node_name = ""
+        if isinstance(node, SnapshotId):
+            node_name = node.name
+        elif isinstance(node, tuple):
+            node_name = node[0]
+        self.node_name = node_name
+
+        # MSSQL returns a `tuple([int] error code, [bytes] error message)`.
+        # TODO: verify return types of other engines
+        error_msg = None
+        for arg in ex.args:
+            if not error_msg and isinstance(arg, (str, bytes)):
+                error_msg = arg.decode(errors="replace") if isinstance(arg, bytes) else arg
+        error_msg = str(ex) if not error_msg else error_msg
+
+        if not isinstance(ex, PythonModelEvalError):
+            error_class = str(ex.__class__).replace("<class '", "").replace("'>", "")
+            error_msg = "  " + error_msg.replace("\n", "\n  ")
```

Comment:
What this indentation for? If it's to format nicely in the console then this logic should be inside the console.